util/docker.go

```diff
@@ -132,19 +133,125 @@ func Login(registry, username, password string) error {
+	return nil
+}
+
+func Tag(origImg, img string) error {
+	var errs bytes.Buffer
+
+	// Run docker tag
+	if img != origImg {
+		fmt.Fprintf(os.Stderr, "INFO: Tagging image %s as %s\n", origImg, img)
+		tagCmd := exec.Command("docker", "tag", origImg, img)
+		tagCmd.Stderr = io.MultiWriter(os.Stderr, &errs)
+		tagCmd.Stdout = os.Stderr
+
+		if err := tagCmd.Run(); err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: Error executing docker tag. %s\n",
+				err.Error())
+		}
+		if errs.String() != "" {
+			fmt.Fprintf(os.Stderr, "ERROR: Error tagging image '%s':\n%s\n", origImg, errs.String())
+			fmt.Fprintf(os.Stderr, "Exiting seed...\n")
+			return errors.New(errs.String())
+		}
+	}
+
+	return nil
+}
+
+func Push(img string) error {
+	var errs bytes.Buffer
+
+	// docker push
+	fmt.Fprintf(os.Stderr, "INFO: Performing docker push %s\n", img)
+	errs.Reset()
+	pushCmd := exec.Command("docker", "push", img)
+	pushCmd.Stderr = io.MultiWriter(os.Stderr, &errs)
+	pushCmd.Stdout = os.Stdout
+
+	// Run docker push
+	if err := pushCmd.Run(); err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: Error executing docker push. %s\n",
 			err.Error())
 		return err
 	}
 
+	// Check for errors. Exit if error occurs
 	if errs.String() != "" {
-		fmt.Fprintf(os.Stderr, "ERROR: Error reading stderr %s\n",
+		fmt.Fprintf(os.Stderr, "ERROR: Error pushing image '%s':\n%s\n", img,
 			errs.String())
+		fmt.Fprintf(os.Stderr, "Exiting seed...\n")
 		return errors.New(errs.String())
 	}
 
-	fmt.Fprintf(os.Stderr, "%s", out.String())
 	return nil
 }
+
+func RemoveImage(img string) error {
+	var errs bytes.Buffer
+
+	fmt.Fprintf(os.Stderr, "INFO: Removing local image %s\n", img)
+	rmiCmd := exec.Command("docker", "rmi", img)
+	rmiCmd.Stderr = io.MultiWriter(os.Stderr, &errs)
+	rmiCmd.Stdout = os.Stdout
+
+	if err := rmiCmd.Run(); err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: Error executing docker rmi. %s\n",
+			err.Error())
+		return err
+	}
+
+	// check for errors on stderr
+	if errs.String() != "" {
+		fmt.Fprintf(os.Stderr, "ERROR: Error removing image '%s':\n%s\n", img,
+			errs.String())
+		fmt.Fprintf(os.Stderr, "Exiting seed...\n")
+		return errors.New(errs.String())
+	}
+
+	return nil
+}
+
+func RestartRegistry() error {
+	fmt.Print("RESTARTING REGISTRY........................\n\n\n\n\n.\n.\n.\n.\n.\n.\n")
+	var errs bytes.Buffer
+
+	fmt.Fprintf(os.Stderr, "INFO: Restarting test registry...\n")
+	cmd := exec.Command("../restartRegistry.sh")
+	cmd.Stderr = io.MultiWriter(os.Stderr, &errs)
+	cmd.Stdout = os.Stdout
+
+	if err := cmd.Run(); err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: Error restarting registry. %s\n",
+			err.Error())
+		return err
+	}
+
+	// check for errors on stderr
+	if errs.String() != "" {
+		fmt.Fprintf(os.Stderr, "ERROR: Error restarting registry. %s\n", errs.String())
+		fmt.Fprintf(os.Stderr, "Exiting seed...\n")
+		return errors.New(errs.String())
+	}
+
+	// wait 100 milliseconds for registry to be running
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Not sure an arbitrary sleep is the best approach here. It would probably be better to just check for 200 OK from the `/v2` api

Reply:
👍